app/src/main/java/com/amaze/filemanager/filesystem/compressed/extractcontents/Extractor.java

```diff
@@ -113,5 +114,11 @@ protected String fixEntryName(String entryName) {
     }
   }
 
-  public class EmptyArchiveNotice extends IOException {}
+  public static class EmptyArchiveNotice extends IOException {}
+
+  public static class BadArchiveNotice extends IOException {
+    public BadArchiveNotice(@Nullable String message) {
```

Comment:
Instead of a string can't we have the original `Exception` saved in the `BadArchiveNotice`?